File: MSI-project/src/app/models/class.ts
import { Rental } from "./rental";
import { UserAccount } from "./UserAccount";

export class Class {
  public get userAccount(): UserAccount {
    return this._userAccount;
  }
  public set userAccount(value: UserAccount) {
    this._userAccount = value;
  }
  public get rental(): Rental {
    return this._rental;
  }
  public set rental(value: Rental) {
    this._rental = value;
  }
  public get level(): string {
    return this._level;
  }
  public set level(value: string) {
    this._level = value;
  }
  public get id(): number {
    return this._id;
  }
  public set id(value: number) {
    this._id = value;
  }
  constructor(
    private _id: number,
    private _level: string,
    private _rental: Rental,
    private _userAccount: UserAccount
  ){}
}

File: MobilneSystemyInformatyczne/ShareWindsurfingEquipment/src/main/java/com/project/ShareWindsurfingEquipment/model/Class.java
package com.project.ShareWindsurfingEquipment.model;

import javax.persistence.*;

@Entity
@Table
public class Class {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String level;

    @ManyToOne
    @JoinColumn(name = "rental_id")
    private Rental rental;

    @ManyToOne
    @JoinColumn(name = "user_account_id")
    private UserAccount userAccount;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getLevel() {
        return level;
    }

    public void setLevel(String level) {
        this.level = level;
    }

    public Rental getRental() {
        return rental;
    }

    public void setRental(Rental rental) {
        this.rental = rental;
    }

    public UserAccount getUserAccount() {
        return userAccount;
    }

    public void setUserAccount(UserAccount userAccount) {
        this.userAccount = userAccount;
    }

    @Override
    public String toString() {
        return "Class{" +
                "id=" + id +
                ", level='" + level + '\'' +
                ", rental=" + rental +
                ", userAccount=" + userAccount +
                '}';
    }
}
